www/main/navigator/shabad/FavShabadIcon.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { useStoreState, useStoreActions } from 'easy-peasy';
+import classNames from '../../common/utils/classnames';
+import { addToFav, fetchFavShabad, removeFromFav } from '../misc/utils';
+
+const remote = require('@electron/remote');
+
+const { i18n } = remote.require('./app');
+
+const FavShabadIcon = ({ paneId }) => {
+  const [isLoading, setLoading] = useState(false);
+  const favBtnRef = useRef(null);
+  const { activeShabadId, activeVerseId, favShabad, pane1, pane2, pane3 } = useStoreState(
+    (state) => state.navigator,
+  );
+
+  const { setFavShabad } = useStoreActions((state) => state.navigator);
+
+  const { userToken } = useStoreState((state) => state.app);
+
+  const [currentShabad, setCurrentShabad] = useState(activeShabadId);
+  const [currentVerse, setCurrentVerse] = useState(activeVerseId);
+  const [favShabadIndex, setFavShabadIndex] = useState(-1);
+
+  useEffect(() => {
+    if (paneId) {
+      switch (paneId) {
+        case 1:
+          setCurrentShabad(pane1.activeShabad);
+          setCurrentVerse(pane1.versesRead[0]);
+          break;
+        case 2:
+          setCurrentShabad(pane2.activeShabad);
+          setCurrentVerse(pane2.versesRead[0]);
+          break;
+        case 3:
+          setCurrentShabad(pane3.activeShabad);
+          setCurrentVerse(pane3.versesRead[0]);
+          break;
+        default:
+          break;
+      }
```

Comment:
The code below is to avoid the violation of DRY principle. What do you think veerji?

```js
const panes = [pane1, pane2, pane3]

if (paneId >= 1 && paneId <= 3) {
    const currentPane = panes[paneId - 1];
    setCurrentShabad(currentPane.activeShabad);
    setCurrentVerse(currentPane.versesRead[0]);
}
```